src/icon/lib/icon.scss

```diff
@@ -0,0 +1,18 @@
+.tbds-icon {
+  --size: 1em;
+
+  fill: currentcolor;
+  height: var(--size);
+}
+
+.tbds-icon--brand {
```

Comment:
Not a huge deal, but this option/modifier feels kinda random. Since we're leaning on `currentcolor`, I'd expect color to come naturally via the parent's `color` value.

Also, since we have brand red as an option, should we also have our brand gray and blue? Either way, we may want to be specific and say `brand-red`, as it's likely we'll always have multiple "brand colors" and people define what that means differently. `brand-red` is just more obvious.